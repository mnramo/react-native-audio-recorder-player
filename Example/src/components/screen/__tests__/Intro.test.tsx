import 'react-native';
import * as React from 'react';
import Intro from '../Intro';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('Intro', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Intro />).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('component test', () => {
    const wrapper = shallow(
      <Intro />,
    );

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
