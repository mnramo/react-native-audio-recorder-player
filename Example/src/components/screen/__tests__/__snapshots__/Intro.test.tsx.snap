// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component test renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(65, 77, 107)",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 58.33333333333333,
        "marginTop": 291.66666666666663,
      }
    }
  >
    DOOBOO NATIVE
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 83.33333333333331,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eee",
          "fontSize": 33.33333333333333,
          "lineHeight": 48,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eee",
          "fontSize": 33.33333333333333,
          "lineHeight": 48,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eee",
          "fontSize": 33.33333333333333,
          "lineHeight": 48,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": 33.33333333333333,
        "position": "absolute",
      }
    }
  >
    <Button
      activeOpacity={0.5}
      imgLeftSrc={
        Object {
          "process": [Function],
        }
      }
      imgLeftStyle={
        Object {
          "height": 49.99999999999999,
          "left": 33.33333333333333,
          "position": "absolute",
          "width": 49.99999999999999,
        }
      }
      indicatorColor="white"
      isDisabled={false}
      isLoading={false}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderColor": "white",
          "borderRadius": 8.333333333333332,
          "borderWidth": 4.166666666666666,
          "height": 108.33333333333331,
          "justifyContent": "center",
          "width": 666.6666666666665,
        }
      }
      textStyle={
        Object {
          "color": "white",
          "fontSize": 29.166666666666664,
        }
      }
    >
      Login
    </Button>
  </View>
</View>
`;

exports[`Intro renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(65, 77, 107)",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 58.33333333333333,
        "marginTop": 291.66666666666663,
      }
    }
  >
    DOOBOO NATIVE
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 83.33333333333331,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eee",
          "fontSize": 33.33333333333333,
          "lineHeight": 48,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eee",
          "fontSize": 33.33333333333333,
          "lineHeight": 48,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#eee",
          "fontSize": 33.33333333333333,
          "lineHeight": 48,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": 33.33333333333333,
        "position": "absolute",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderColor": "white",
            "borderRadius": 8.333333333333332,
            "borderWidth": 4.166666666666666,
            "height": 108.33333333333331,
            "justifyContent": "center",
            "width": 666.6666666666665,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 49.99999999999999,
              "left": 33.33333333333333,
              "position": "absolute",
              "width": 49.99999999999999,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 29.166666666666664,
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
